src/main/java/com/epam/aidial/core/config/FileConfigStore.java

```diff
@@ -83,21 +80,20 @@ private void load(boolean fail) {
                 baseAssistant.setName(ASSISTANT);
                 baseAssistant.setEndpoint(assistants.getEndpoint());
                 baseAssistant.setFeatures(assistants.getFeatures());
-                associateDeploymentWithApiKey(config, baseAssistant);
                 assistants.getAssistants().put(ASSISTANT, baseAssistant);
             }
 
             for (Map.Entry<String, Application> entry : config.getApplications().entrySet()) {
                 String name = entry.getKey();
                 Application application = entry.getValue();
                 application.setName(name);
-                associateDeploymentWithApiKey(config, application);
             }
 
             for (Map.Entry<String, Key> entry : config.getKeys().entrySet()) {
                 String key = entry.getKey();
                 Key value = entry.getValue();
                 value.setKey(key);
+                apiKeyStore.addProjectKey(value);
```

Comment:
When do you remove? e.g. a config changes and does not have old keys.

Reply:
good catch. I'll remove all project keys before adding new ones from the updated config.

Reply:
fixed